src/metabase/public_settings.clj

```diff
@@ -147,6 +147,15 @@
   :default    false
   :visibility :authenticated)
 
+(defsetting embedding-secret-key
+  (deferred-tru "Secret key used to sign JSON Web Tokens for requests to `/api/embed` endpoints.")
+  :visibility :admin
+  :setter (fn [new-value]
+            (when (seq new-value)
+              (assert (u/hexadecimal-string? new-value)
+                (tru "Invalid embedding-secret-key! Secret key must be a hexadecimal-encoded 256-bit key (i.e., a 64-character string).")))
+            (setting/set-string! :embedding-secret-key new-value)))
+
```

Comment:
A secret key is DEFINITELY not a public setting, and since this is visibility `:admin` it's not public anyway, so moving it to this namespace doesn't really make sense IMO

Reply:
I had to move it here, otherwise my tests would fail.

Because I didn't have a dependency on the namespace where it was in the `:require` section, I got an error that the setting did not exist. I couldn't add a reference to that namespace in `:require` because it would get deleted as soon as I did a `cljr-clean-ns` or the linter checked for unused `:require` statements.